Run PHPStan and fix issues

pamtbaau opened the pull request:
I've used the PHPstan config from Grav and ran on level 8. 

Fixed type annotations and bugs, hardened some code and performed some cleanup on the following code:
- classes/Data/*
- classes/RequestHandler.php
- pushy.php (fixed my additions only)

pamtbaau committed:
Run PHPStan and fix issues

hughbris commented on classes/Data/ChangedItems.php:
```diff
@@ -10,15 +10,21 @@
  */
 class ChangedItems implements ArrayAccess, Countable, Iterator, JsonSerializable
 {
-    /** @var array<string, ChangedItems> */
```
That's funny, I noticed this a few days ago but forgot to raise it with you.

hughbris commented on classes/Data/ChangedItems.php:
```diff
@@ -27,7 +33,7 @@ public function toArray(): array
             $changedItems[$url] = (array) $changedItem;
         }
 
-        return $changedItem;
```
.. or maybe it was this one

pamtbaau replied:
Or maybe both... 

Some say you shouldn't use plural forms of variables by adding an `s`. Too prone to mistakes...

Although strong typed languages would have prevented the mistake.

pamtbaau committed:
Cleanup some more

- Remove unused leftovers from copy/paste
- Fixed last few level 9 phpstan issues

pamtbaau commented:
Fixed last few phpstan level 9 issues and cleaned up unused code from copy/paste.

hughbris reviewed (approved):
Looks good
